tests/checks/test_check_ConfirmFileExists.py

```diff
@@ -0,0 +1,142 @@
+    assert "usage:" in captured.err
+    counted_newlines = captured.err.count("\n")
+    assert counted_newlines == 2
+
+
+@pytest.mark.parametrize(
+    "commandline_arguments",
+    [
+        (["--file", "filename", "--directory", "directoryname"]),
+        (["--directory", "directoryname", "--file", "filename"]),
+    ],
+)
+def test_required_commandline_arguments_can_parse(commandline_arguments, not_raises):
+    """Check that correct optional command-line arguments check correctly."""
+    with not_raises(SystemExit):
+        _ = check_ConfirmFileExists.parse(commandline_arguments)
+
+
+@pytest.mark.parametrize(
+    "commandline_arguments",
+    [
+        (["--file", "filename", "--directory", "directoryname"]),
+        (["--directory", "directoryname", "--file", "filename"]),
+    ],
+)
+def test_optional_commandline_arguments_can_parse_created_parser(
+    commandline_arguments, not_raises
+):
+    """Check that correct optional command-line arguments check correctly."""
+    with not_raises(SystemExit):
+        parser = check_ConfirmFileExists.get_parser()
+        _ = check_ConfirmFileExists.parse(commandline_arguments, parser)
+
+
+@pytest.mark.parametrize(
+    "commandline_arguments, chosen_file, containing_directory, expected_result",
+    [
+        (
+            [
+                "ConfirmFileExists",
+                "--file",
+                "file_to_find",
+                "--directory",
+                "containing_directory",
+            ],
+            "file_to_find",
+            "containing_directory",
+            True,
+        )
+    ],
+)
+def test_act_produces_output(
+    commandline_arguments, chosen_file, containing_directory, expected_result, tmpdir
+):
+    """Check that using the check produces output."""
+    testargs = [os.getcwd()]
+    with patch.object(sys, "argv", testargs):
+        new_file = tmpdir.mkdir(containing_directory).join(chosen_file)
+        new_file.write("\\begin{document} hello! \\end{document}")
+        assert new_file.read() == "\\begin{document} hello! \\end{document}"
+        assert len(tmpdir.listdir()) == 1
+        overall_directory = (
+            tmpdir.dirname + "/" + tmpdir.basename + "/" + containing_directory
+        )
+        commandline_arguments = [
+            "ConfirmFileExists",
+            "--file",
+            "file_to_find",
+            "--directory",
+            overall_directory,
+        ]
+        parsed_arguments, remaining_arguments = arguments.parse(commandline_arguments)
+        args_verified = arguments.verify(parsed_arguments)
+        assert args_verified is True
+        external_checker_directory = checkers.get_checker_dir(parsed_arguments)
+        checker_source = checkers.get_source([external_checker_directory])
+        check_name = checkers.get_chosen_check(parsed_arguments)
+        check_file = checkers.transform_check(check_name)
+        check_exists = checkers.verify_check_existence(check_file, checker_source)
```

Comment:
There should probably be a utility function or fixture that loads a check specifically for the test suite, so that these details don't complicate the test. Another future idea :)

Reply:
Since this suggestion would require me to change test code and _not_ production code-under-test, I went ahead and fully implemented a solution. The implementation is slightly complicated by the fact that Pytest fixtures cannot accept parameters. However, I have an approach that creates a fixture that returns a function that can then accept the parameters.

Since I have fully resolved this issue I will mark the conversation as resolved.